controllers/common/runtime_controller.go

```diff
@@ -0,0 +1,144 @@
+	log := logger.FromContext(ctx)
+	log.Info("Reconciling for KogitoRuntime")
+
+	// create kogitoContext
+	kogitoContext := operator.Context{
+		Client:  r.Client,
+		Log:     log,
+		Scheme:  r.Scheme,
+		Version: r.Version,
+	}
+
+	// fetch the requested instance
+	runtimeHandler := r.RuntimeHandler(kogitoContext)
+	instance, err := runtimeHandler.FetchKogitoRuntimeInstance(req.NamespacedName)
+	if err != nil {
+		return
+	}
+	if instance == nil {
+		log.Debug("KogitoRuntime instance not found")
+		return
+	}
+
+	rbacHandler := infrastructure.NewRBACHandler(kogitoContext)
+	if err = rbacHandler.SetupRBAC(req.Namespace); err != nil {
+		return
+	}
+
+	supportingServiceHandler := r.SupportServiceHandler(kogitoContext)
+	deploymentHandler := NewRuntimeDeployerHandler(kogitoContext, instance, supportingServiceHandler, runtimeHandler)
+	definition := kogitoservice.ServiceDefinition{
+		Request:            req,
+		DefaultImageTag:    infrastructure.LatestTag,
+		SingleReplica:      false,
+		OnDeploymentCreate: deploymentHandler.OnDeploymentCreate,
+		CustomService:      true,
+	}
+	infraHandler := app.NewKogitoInfraHandler(kogitoContext)
+	err = kogitoservice.NewServiceDeployer(kogitoContext, definition, instance, infraHandler).Deploy()
+	if err != nil {
+		return infrastructure.NewReconciliationErrorHandler(kogitoContext).GetReconcileResultFor(err)
+	}
+
+	protoBufHandler := shared.NewProtoBufHandler(kogitoContext, supportingServiceHandler)
+	err = protoBufHandler.MountProtoBufConfigMapOnDataIndex(instance)
+	if err != nil {
+		log.Error(err, "Fail to mount Proto Buf config map of Kogito runtime on DataIndex")
+		return infrastructure.NewReconciliationErrorHandler(kogitoContext).GetReconcileResultFor(err)
+	}
+
+	log.Debug("Finish reconciliation", "requeue", result.Requeue, "requeueAfter", result.RequeueAfter)
+	return
+}
+
+// SetupWithManager registers the controller with manager
+func (r *KogitoRuntimeReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	pred := predicate.Funcs{
+		// Don't watch delete events as the resource removals will be handled by its finalizer
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.GetDeletionTimestamp().IsZero()
+		},
+	}
+	b := ctrl.NewControllerManagedBy(mgr).
+		For(r.ReconcilingObject, builder.WithPredicates(pred)).
+		Owns(&corev1.Service{}).Owns(&appsv1.Deployment{}).Owns(&corev1.ConfigMap{})
+
+	//infraHandler := &handler.EnqueueRequestForOwner{IsController: false, OwnerType: &v1beta1.KogitoRuntime{}}
+	//infraPred := predicate.Funcs{
+	//	UpdateFunc: func(e event.UpdateEvent) bool {
+	//		return reflect.DeepEqual(e.ObjectNew.GetOwnerReferences(), e.ObjectOld.GetOwnerReferences())
+	//	},
+	//}
+	//b.Watches(&source.Kind{Type: &v1beta1.KogitoInfra{}}, infraHandler, builder.WithPredicates(infraPred))
```

Comment:
Why is this code commented out?

Reply:
Its not required anymore because I had removed KogitoInfra ownership from KogitoRuntime and KogitoSupportingServices.